orm/versioning_bucket.go

```diff
@@ -0,0 +1,109 @@
+package orm
+
+import (
+	"fmt"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+type VersioningBucket struct {
+	IDGenBucket
+}
+
+func WithVersioning(b IDGenBucket) VersioningBucket {
+	return VersioningBucket{b}
+}
+
+func (b VersioningBucket) GetLatestVersion(db weave.ReadOnlyKVStore, id []byte) (Object, error) {
+	idWithoutVersion := &VersionedIDRef{ID: id}
+	prefix, err := idWithoutVersion.Marshal()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal versioned ID ref")
+	}
+	dbKeyLength := len(b.DBKey(prefix)) - len(prefix)
+	fmt.Printf("searching with: #%v\n", prefix)
+	matches, err := b.Query(db, weave.PrefixQueryMod, prefix)
+	if err != nil {
+		return nil, errors.Wrap(err, "prefix query")
+	}
+	// find highest version for that ID
+	var highestVersion VersionedIDRef
+	var found weave.Model
+	for _, m := range matches {
+		var vID VersionedIDRef
+		idData := m.Key[dbKeyLength:]
+		fmt.Printf("found : #%v\n", idData)
+
+		if err := vID.Unmarshal(idData); err != nil {
```

Comment:
So you go through all versions, decode them with protobuf, then compare.

Pretty intensive operation. Just store the version bytes in a way that lexographical ordening works for you and you can compare the raw encoded bytes (and just decode one)

Reply:
The `VersionedIDRef` object is supposed to be used as reference in the Proposal model to point to the Electorate. Because of this I need it as primary key. 
I am not fully happy with this impl of `GetLatestVersion` though. Will give it another thought.